src/analyses/dependence_graph.cpp

```diff
@@ -26,36 +26,20 @@ bool dep_graph_domaint::merge(
   goto_programt::const_targett from,
   goto_programt::const_targett to)
 {
-  bool changed=has_values.is_false();
-  has_values=tvt::unknown();
+  bool changed=has_values.is_false() || has_changed;
 
-  depst::iterator it=control_deps.begin();
-  for(const auto &c_dep : src.control_deps)
+  if((has_values.is_false() || has_changed) &&
```

Comment:
has_changed already includes the case that `has_values.is_false()`? I this is just `has_changed && data_deps.empty()`